Tests/UnitTests/Core/TrackFitting/Gx2fTests.cpp

```diff
@@ -0,0 +1,352 @@
+                                           10_degree, 80_degree, 1_GeV, 1_e);
+  //  auto startParametersFit = parametersMeasurements;
+  // Context objects
+  Acts::GeometryContext geoCtx;
+  Acts::MagneticFieldContext magCtx;
+  // Acts::CalibrationContext calCtx;
+  std::default_random_engine rng(42);
+
+  MeasurementResolution resPixel = {MeasurementType::eLoc01, {25_um, 50_um}};
+  // MeasurementResolution resStrip0 = {MeasurementType::eLoc0, {100_um}};
+  // MeasurementResolution resStrip1 = {MeasurementType::eLoc1, {150_um}};
+  MeasurementResolutionMap resolutions = {
+      {Acts::GeometryIdentifier().setVolume(0), resPixel}};
+
+  // simulation propagator
+  using SimPropagator =
+      Acts::Propagator<Acts::StraightLineStepper, Acts::Navigator>;
+  SimPropagator simPropagator = makeStraightPropagator(detector.geometry);
+  auto measurements = createMeasurements(
+      simPropagator, geoCtx, magCtx, parametersMeasurements, resolutions, rng);
+  auto sourceLinks = prepareSourceLinks(measurements.sourceLinks);
+  std::cout << "sourceLinks.size() = " << sourceLinks.size() << std::endl;
+
+  BOOST_REQUIRE_EQUAL(sourceLinks.size(), nSurfaces);
+
+  std::cout << "\n*** Start fitting ***\n" << std::endl;
+
+  std::cout << "\n*** startParameter unsmeared: ***\n"
+            << parametersMeasurements << std::endl;
+  std::cout << "\n*** startParameter fit: ***\n"
+            << startParametersFit << std::endl;
+  const Surface* rSurface = &parametersMeasurements.referenceSurface();
+
+  Navigator::Config cfg{detector.geometry};
+  cfg.resolvePassive = false;
+  cfg.resolveMaterial = true;
+  cfg.resolveSensitive = true;
+  Navigator rNavigator(cfg);
+  // Configure propagation with deactivated B-field
+  auto bField = std::make_shared<ConstantBField>(Vector3(0., 0., 0.));
+  using RecoStepper = EigenStepper<>;
+  RecoStepper rStepper(bField);
+  using RecoPropagator = Propagator<RecoStepper, Navigator>;
+  RecoPropagator rPropagator(rStepper, rNavigator);
+
+  using Gx2Fitter =
+      Experimental::Gx2Fitter<RecoPropagator, VectorMultiTrajectory>;
+
+  Gx2Fitter Fitter(rPropagator);
+
+  Experimental::Gx2FitterExtensions<VectorMultiTrajectory> extensions;
+  extensions.calibrator
+      .connect<&testSourceLinkCalibrator<VectorMultiTrajectory>>();
+
+  MagneticFieldContext mfContext;
+  CalibrationContext calContext;
+
+  const Experimental::Gx2FitterOptions gx2fOptions(
+      tgContext, mfContext, calContext, extensions, PropagatorPlainOptions(),
+      rSurface, false, false, FreeToBoundCorrection(false), 5);
+
+  Acts::TrackContainer tracks{Acts::VectorTrackContainer{},
+                              Acts::VectorMultiTrajectory{}};
+
+  // Fit the track
+  auto res = Fitter.fit(sourceLinks.begin(), sourceLinks.end(),
+                        startParametersFit, gx2fOptions, tracks);
+
+  BOOST_REQUIRE(res.ok());
+
+  auto& track = *res;
+  BOOST_CHECK_EQUAL(track.tipIndex(), Acts::MultiTrajectoryTraits::kInvalid);
```

Comment:
Also just a question: Why should the tip index be invalid?

Reply:
Because the current implementation defaults most of the contents of the returned trackContainer. After I add this (and `nHoles`, `nMeasurements`, ...) I would change to `BOOST_CHECK_NE()` like in the tests for the other fitters.

Reply:
ok, makes sense!